otx/cli/utils/multi_gpu.py

```diff
@@ -0,0 +1,245 @@
+
+        Returns:
+            bool:
+                whether multi GPU training is available.
+        """
+        return (
+            len(self._gpu_ids) > 1
+            and not template.task_type.is_anomaly
+            and template.task_type not in (TaskType.ACTION_CLASSIFICATION, TaskType.ACTION_DETECTION)
+        )
+
+    def setup_multi_gpu_train(
+        self,
+        output_path: str,
+        optimized_hyper_parameters: Optional[ConfigurableParameters] = None,
+    ):
+        """Carry out what should be done to run multi GPU training.
+
+        Args:
+            output_path (str): output path where task output are saved.
+            optimized_hyper_parameters (ConfigurableParameters or None): hyper parameters reflecting HPO result.
+        """
+        if optimized_hyper_parameters is not None:  # if HPO is executed, optimized HPs are applied to child processes
+            self._set_optimized_hp_for_child_process(optimized_hyper_parameters)
+
+        self._processes = self._spawn_multi_gpu_processes(output_path)
+
+        signal.signal(signal.SIGINT, self._terminate_signal_handler)
+        signal.signal(signal.SIGTERM, self._terminate_signal_handler)
+
+        self.initialize_multigpu_train(0, self._gpu_ids, self._multi_gpu_port)
+
+        threading.Thread(target=self._check_child_processes_alive, daemon=True).start()
+
+    def finalize(self):
+        """Join all child processes."""
+        if self._processes is not None:
+            for p in self._processes:
+                p.join()
+
+    @staticmethod
+    def initialize_multigpu_train(rank: int, gpu_ids: List[int], multi_gpu_port: str):
+        """Initilization for multi GPU training.
+
+        Args:
+            rank (int): index of multi GPU processes.
+            gpu_ids (List[int]): list including which GPU indeces will be used.
+            multi_gpu_port (str): port for communication between multi GPU processes.
+        """
+        os.environ["MASTER_ADDR"] = "localhost"
+        os.environ["MASTER_PORT"] = multi_gpu_port
+        torch.cuda.set_device(gpu_ids[rank])
+        dist.init_process_group(backend="nccl", world_size=len(gpu_ids), rank=rank)
+        print(f"dist info world_size = {dist.get_world_size()}, rank = {dist.get_rank()}")
+
+    @staticmethod
+    def run_child_process(train_func: Callable, rank: int, gpu_ids: List[int], output_path: str, multi_gpu_port: str):
+        """Function for multi GPU child process to execute.
+
+        Args:
+            train_func (Callable): model training function.
+            rank (int): index of multi GPU processes.
+            gpu_ids (List[int]): list including which GPU indeces will be used.
+            output_path (str): output path where task output are saved.
+            multi_gpu_port (str): port for communication between multi GPU processes.
+        """
+        gpus_arg_idx = sys.argv.index("--gpus")
+        for _ in range(2):
```

Comment:
is there any better way to change this line? I can't easily understand the meaning of 2.
Or adding a description? ``` for in range(2) # meaning of 2```

Reply:
It means that it removes both argument key and value.
I think I can write that code as bellow. what's your opinion about this? is it better?
```
sys.argv.pop(gpus_arg_idx+1)
sys.argv.pop(gpus_arg_idx)
```

Reply:
> It means that it removes both argument key and value. I think I can write that code as bellow. what's your opinion about this? is it better?
> 
> ```
> sys.argv.pop(gpus_arg_idx+1)
> sys.argv.pop(gpus_arg_idx)
> ```

Ah, I see. let keep the current code :)